autolab/Autolab
# Add filename, line number, better collapsibility to annotations

## Description

- Allow multiple problems open at once, fix stutter for initial open
- Add file name to annotations
- Add line number to annotations
- Add open / close arrows next to category name
- Use jquery load to refresh annotations
- Make scores in annotations and grades tabs 2 d.p.
- Make default annotation score 0, set type to number
- Remove some dead code
- Validate against missing annotation scores
- Fix annotation-box badge color updates for score 0 annotations (previously, if you update an annotation that had score 0, the "badge" turns green instead of staying gray)

## Motivation and Context

Initial work for generalized feedback.

## How Has This Been Tested?
- Create an assessment that accepts tar files
- Submit a tar file (e.g. hello.tar)
- Create annotations, taking note that
  - Annotations appearance is consistent after updates (add / delete)
  - Annotations are still clickable (and bring you to its location) after updates
  - Annotations are grouped by filename
  - Line numbers appear for annotations
  - Multiple categories can be open at once
  - No more stuttering when collapsing a category
  - It is no longer possible to create an annotation without a score
  - Creating, then updating annotations with -ve, 0, +ve scores preserve the "badge colors" in annotation boxes as expected

## Types of changes

## Checklist:
